sdk/storage/AzureStorageBlob/Source/DataStructures/TransferManager.swift

```diff
@@ -0,0 +1,103 @@
+// --------------------------------------------------------------------------
+//
+// Copyright (c) Microsoft Corporation. All rights reserved.
+//
+// The MIT License (MIT)
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the ""Software""), to
+// deal in the Software without restriction, including without limitation the
+// rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
+// sell copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+// FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+// IN THE SOFTWARE.
+//
+// --------------------------------------------------------------------------
+
+import AzureCore
+import CoreData
+
+// MARK: Protocols
+
+public protocol TransferManager: ResumableOperationQueueDelegate {
+    // MARK: Properties
+
+    var reachability: ReachabilityManager? { get }
+    var persistentContainer: NSPersistentContainer? { get }
+    static var shared: Self { get }
+    var logger: ClientLogger { get set }
+
+    // MARK: Storage Methods
+
+    // func upload(_ url: URL) -> Transferable
```

Comment:
Added a TODO and comment. Essentially, once the signatures that exist in the TM implementation are reviewed and approved, we can uplevel them into the protocol definition. They definitely won't be these signatures though, since "Transferable" is no longer a thing.